### PR TITLE
Emit deprecation build operation progress notifications for deprecated usages on unmanaged threads

### PR DESCRIPTION
**Pull request by ldaley**

### CODE REVIEW

**Comment by ldaley** [comment]
@bot-gradle test RFM

**Comment by ldaley** [comment]
@adammurdoch this change allows scans to observe deprecations occurring when we don't know what the current build op is, such as when unmanaged threads are used.

**Comment by ldaley** [comment]
@lptr this should address your comments about naming from my previous PR.

**Comment by ldaley** [comment]
@bot-gradle test RFM

**Comment by ldaley** [comment]
@bot-gradle test RFM

**Comment by ldaley** [comment]
@bot-gradle test RFM

File: subprojects/logging/src/integTest/groovy/org/gradle/DeprecatedUsageBuildOperationProgressIntegrationTest.groovy [hunk 1]
```diff
@@ -70,10 +70,19 @@ class DeprecatedUsageBuildOperationProgressIntegrationTest extends AbstractInteg
                 }
             }
 
+            task t3 {
+                doLast {
+                    def latch = new java.util.concurrent.CountDownLatch(1)
```
Comment (adammurdoch) [inline subprojects/logging/src/integTest/groovy/org/gradle/DeprecatedUsageBuildOperationProgressIntegrationTest.groovy:75]: seems to be unused? Was this going to be used to synchronise with the other thread?

Comment (ldaley) [reply subprojects/logging/src/integTest/groovy/org/gradle/DeprecatedUsageBuildOperationProgressIntegrationTest.groovy:75]: Forgot to use it. Added.

**Comment by ldaley** [comment]
@bot-gradle test RFM

**Comment by lptr** [comment]
@ldaley I just took another look because I'd like to move most of the build operation-related logic to a new subproject (so we can share it with Maven caching). And it looks to me that the emitter implementation should be in `:core`, while only a thin interface should be in `:base-services` (later in `:build-operations`) that can be injected into the deprecation logger.

Can you please make this change? It would also allow the build operations package not to depend on `@ServiceScope` etc.

**Comment by lptr** [comment]
@ldaley is this something you want to get merged in 6.7 still? If so, how can we make that happen?

**Comment by big-guy** [comment]
Please reopen against 6.8 RC1 if we need this still.

**Pull request closed without merging**